docs/content/preview/api/ysql/the-sql-language/statements/ddl_alter_table.md

```diff
@@ -198,6 +198,50 @@ It quietly succeeds. Now `\d children` shows that the foreign key constraint `ch
 
 Add the specified constraint to the table.
 
+#### [*alter_column_type*]
+
+Change the type of an existing column. If data on disk is required to change, a full table rewrite is needed.
+
+The new column type will be applied automatically to indexes and simple table constraints by reanalyzing the initially provided expression.
+
+If the optional `COLLATE` clause is not specified, the default collation for the new column type will be used.
+
+If the optional `USING` clause is not provided, the default conversion for the new column value will be the same as an assignment cast from the old type to the new type.
+
+A `USING` clause must be included when there is no implicit or assignment cast available from the old type to the new type.
+
+##### Alter type without table-rewrite
+
+If the change doesn't require data on disk to change, these semantics apply:
+
+- Concurrent DMLs to the table can go through.
+
+Example:
+
+```sql
+CREATE TABLE test (id BIGSERIAL PRIMARY KEY, a VARCHAR(50));
+ALTER TABLE test ALTER COLUMN a TYPE VARCHAR(51);
+```
+
+##### Alter type with table rewrite
+
+If the change requires data on disk to change, a full table rewrite will be done and the following semantics apply:
+- This action is unsafe, so concurrent DMLs may be dropped and should not be performed while an alter column type operation is happening.
+- The action creates an entirely new table under the hood, and concurrent DMLs may not be reflected in the new table which can lead to correctness issues.
+- This action is not compatible with: partitioned tables, tables with rules, and tables with CDC streams.
+- If the operation fails, it is possible that the existing table is renamed in DocDB. This may lead to issues like yb-admin commands that take table name.
+- If the operation fails, a new dangling table may exist.
+- Altering the data type of a foreign key column is not supported.
+- If there might be concurrent DMLs the user can rename the table first to fail the DMLs, alter the column data type, and then rename the table again.
+
+Example:
+
+```sql
+CREATE TABLE test (id BIGSERIAL PRIMARY KEY, a VARCHAR(50));
+ALTER TABLE test ALTER COLUMN a TYPE VARCHAR(40);
+ALTER TABLE test ALTER COLUMN a SET DATA TYPE BIGINT USING a::BIGINT;
```

Comment:
Add a `USING <function>` example

Reply:
not done

Reply:
That is a cast, not function

Reply:
Added in https://github.com/yugabyte/yugabyte-db/pull/17470/commits/31861f1de0ee4de05b8b1e3164d5ef104e02914c